compiler/src/dmd/dsymbolsem.d

```diff
@@ -4146,7 +4146,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             else if (dim == 0 && tf.parameterList.varargs != VarArg.none) // allow varargs only ctor
             {
             }
-            else if (dim && tf.parameterList[0].defaultArg)
+            else if (dim && !tf.parameterList.hasArgsWithoutDefault)
             {
                 // if the first parameter has a default argument, then the rest does as well
```

Comment:
I suspect that comment is no longer relevant ;)

Reply:
Good catch :)